### PR TITLE
Ports Fission360 and Lifeline program from /tg/

### PR DESCRIPTION
**Pull request by BrumalLake**
## About The Pull Request
Ports [tgstation#51152](https://github.com/tgstation/tgstation/pull/51152), [tgstation#51386](https://github.com/tgstation/tgstation/pull/51386), and [tgstation#52783](https://github.com/tgstation/tgstation/pull/52783)
Fission360 - emagged modular pc program that works both as a pinpointer that lets you track nuclear related devices and gives a grid view in modular computers
Nukies get it with a tablet preinstalled instead of a pinpointer
image

Lifeline - Suit sensor pinpointer with grid view
image

image

image

## Why It's Good For The Game
More features and versatility for modular computers

## Changelog
:cl: lordScrubling, zxaber
add: Fission360 and Lifeline program for modular computers
add: Nuclear operatives start with a tablet instead of a pinpointer
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
DNM pending TGUI update

**Comment by francinum** [comment]
Pics

**Comment by BrumalLake** [comment]
> 
> 
> Pics

original pr? they look identical

File: code/modules/modular_computers/file_system/programs/radar.dm [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+	if(!humanoid || !istype(humanoid))
+		return FALSE
+	if(..())
+		if(humanoid in SSnanites.nanite_monitored_mobs)
+			return TRUE
+		if(istype(humanoid.w_uniform, /obj/item/clothing/under))
+			var/obj/item/clothing/under/uniform = humanoid.w_uniform
+			if(uniform.has_sensor && uniform.sensor_mode >= SENSOR_COORDS) // Suit sensors must be on maximum
+				return TRUE
+	return FALSE
```
Comment (PowerfulBacon) [inline code/modules/modular_computers/file_system/programs/radar.dm]: This needs to be the same as the crew monitor and check things like is_jammed()

**Review by PowerfulBacon — DISMISSED** [review]

File: code/modules/modular_computers/file_system/programs/radar.dm (hunk 1 above)
Comment (BrumalLake) [reply code/modules/modular_computers/file_system/programs/radar.dm]: signal jammers suck

File: code/modules/modular_computers/file_system/programs/radar.dm [hunk 2]
```diff
@@ -0,0 +1,295 @@
+/datum/computer_file/program/radar //generic parent that handles most of the process
+	filename = "genericfinder"
+	filedesc = "debug_finder"
+	ui_header = "borg_mon.gif" //DEBUG -- new icon before PR
```
Comment (Archanial) [inline code/modules/modular_computers/file_system/programs/radar.dm:6]: :flushed:

File: code/modules/modular_computers/file_system/programs/radar.dm [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+/datum/computer_file/program/radar/proc/track()
+	var/atom/movable/signal = find_atom()
+	if(!trackable(signal))
+		return
+
+	var/turf/here_turf = (get_turf(computer))
+	var/turf/target_turf = (get_turf(signal))
+	var/userot = FALSE
+	var/rot = 0
+	var/pointer="crosshairs"
```
Comment (Archanial) [inline code/modules/modular_computers/file_system/programs/radar.dm]:
```suggestion
	var/pointer = "crosshairs"
```

File: code/modules/modular_computers/file_system/programs/radar.dm [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+	arrowstyle = "ntosradarpointerS.png"
+	pointercolor = "red"
+
+/datum/computer_file/program/radar/fission360/find_atom()
+	return locate(selected) in GLOB.poi_list
+
+/datum/computer_file/program/radar/fission360/scan()
+	if(world.time < next_scan)
+		return
+	next_scan = world.time + (2 SECONDS)
```
Comment (Archanial) [inline code/modules/modular_computers/file_system/programs/radar.dm]: Wouldn't it be better to use our cooldowns?

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/modules/modular_computers/file_system/programs/radar.dm (hunk 4 above)
Comment (BrumalLake) [reply code/modules/modular_computers/file_system/programs/radar.dm]: what do you mean

**Comment by ike709** [comment]
Yolomerge because I want to test the TGUI conflict fixer.

**Pull request merged**